### PR TITLE
Replace fse.exists with fs.promises.access

### PR DESCRIPTION
**Pull request by sdas13**
Addresses: #8605

### CODE REVIEW

**Review by medikoo — CHANGES_REQUESTED** [review]
@sdas13 great thanks for that, still there was no plan to introduce such refactor.

All `*sync` methods are scheduled to be replaced with _async_ variants (see https://github.com/serverless/serverless/issues/8604)

For `fse` methods replacement, please stick to methods listed in description of https://github.com/serverless/serverless/issues/8605

Thank you!

**Comment by sdas13** [comment]
Hello @medikoo, thanks for the feedback. Will revert to `fse.existsSync` methods. 

In the codebase, I could find one instance of `fse.exists` being used. Before replacing it with `fs.promises.exists` method, I can see the `exists` method is [deprecated](https://nodejs.org/api/fs.html#fs_fs_exists_path_callback).
Also the `fs.promises` Api doesn't have `exists` method. 

Should I use `fs.promises.access()` instead?

**Comment by medikoo** [comment]
> Should I use fs.promises.access() instead?

@sdas13 that's a very good question, and yes in this case `fs.promises.access()` looks as most appropriate replacement

File: lib/plugins/interactiveCli/initializeService.js [hunk 1]
```diff
@@ -43,8 +43,13 @@ const projectNameInput = async (workingDir) =>
...
         }
-        if (await fse.exists(join(workingDir, input))) return `Path ${input} is already taken`;
-        return true;
+
+        try {
+          await fs.promises.access(join(workingDir, input));
+          return `Path ${input} is already taken`;
+        } catch {
+          return true;
+        }
```
Comment (sdas13) [inline lib/plugins/interactiveCli/initializeService.js:52]: Hey @medikoo, these statements can also be written like this
`fs.promises.access(join(workingDir, input))
      .then(() => Path ${input} is already taken)
      .catch(() => true)`
Which one do you prefer?

Comment (medikoo) [reply lib/plugins/interactiveCli/initializeService.js:52]: @sdas13 we definitely should stick to async/await syntax. It's way more readable and maintanable

File: lib/utils/createFromTemplate.js [hunk 2]
```diff
@@ -43,18 +44,18 @@ module.exports = (templateName, destPath, options = {}) =>
         }
         resolve(
           BbPromise.all([
-            new BbPromise((gitignoreRenameResolve, gitIgnoreRenameReject) => {
+            new BbPromise(async (gitignoreRenameResolve, gitIgnoreRenameReject) => {
```
Comment (medikoo) [inline lib/utils/createFromTemplate.js:47]: This creates an orphaned promise. Let's not do that

File: lib/utils/createFromTemplate.js [hunk 3]
```diff
@@ -43,18 +44,18 @@ module.exports = (templateName, destPath, options = {}) =>
...
               const gitignorePath = join(destPath, 'gitignore');
-              exists(gitignorePath, (hasGitignore) => {
-                if (!hasGitignore) {
-                  gitignoreRenameResolve();
-                  return;
-                }
+
+              try {
+                await fs.promises.access(gitignorePath);
```
Comment (medikoo) [inline lib/utils/createFromTemplate.js:51]: In this scenario we need to go with regular `promise.then` syntax.

Alternatively we may first refactor this function to async/await (in other PR), and then go back to this PR and rely on async/await

**Review by medikoo — CHANGES_REQUESTED** [review]
Thank you @sdas13 ! It looks good, we just need to update one reference

File: lib/utils/createFromTemplate.js [hunk 4]
```diff
@@ -45,16 +46,16 @@ module.exports = (templateName, destPath, options = {}) =>
...
+
+              fs.promises
+                .access(gitignorePath)
+                .then(() => {
+                  rename(gitignorePath, join(destPath, '.gitignore'), (renameError) => {
+                    if (renameError) gitIgnoreRenameReject(renameError);
+                    else gitignoreRenameResolve();
+                  });
+                })
+                .catch(gitignoreRenameResolve);
```
Comment (medikoo) [inline lib/utils/createFromTemplate.js:58]: Let's replace this whole `new BbPromise` construct with:

```javascript
fs.promises.access(gitignorePath).then(
  () => rename(gitignorePath, join(destPath, '.gitignore')),
  () => {}
)
```

**Review by medikoo — CHANGES_REQUESTED** [review]
Thank you @sdas13, it's way better. Still I've proposed additional improvement that cleans the second (unnecessarily verbose) construct

File: lib/utils/createFromTemplate.js (hunk 4 above)
Comment (sdas13) [reply lib/utils/createFromTemplate.js:58]: got it

**Review by medikoo — APPROVED** [review]
Thank you @sdas13 ! Looks great

**Pull request merged**